src/compiler/transformers/module/module.ts

```diff
@@ -2343,10 +2375,15 @@ export function transformModule(context: TransformationContext): (x: SourceFile
                 else if (isImportSpecifier(importDeclaration)) {
                     const name = importDeclaration.propertyName || importDeclaration.name;
                     return setTextRange(
-                        factory.createPropertyAccessExpression(
-                            factory.getGeneratedNameForNode(importDeclaration.parent?.parent?.parent || importDeclaration),
-                            factory.cloneNode(name),
-                        ),
+                        name.kind === SyntaxKind.StringLiteral
+                            ? factory.createElementAccessExpression(
+                                factory.getGeneratedNameForNode(importDeclaration.parent?.parent?.parent || importDeclaration),

```

Comment:
This line can be extracted into an independent variable and reused on L2384 to cut down on nesting.